include/internal/hashtable.h

```diff
@@ -221,7 +244,10 @@ pfx ossl_unused vtype *ossl_unused ossl_ht_##name##_##vtype##_get(HT *h,       \
     vv = ossl_ht_get(h, key);                                                  \
     if (vv == NULL)                                                            \
         return NULL;                                                           \
-    *v = ossl_rcu_deref(&vv);                                                  \
+    if (!ossl_ht_get_cfg_no_rcu(h))                                            \
+        *v = ossl_rcu_deref(&vv);                                              \
+    else                                                                       \
+        *v = vv;                                                               \
```

Comment:
Instead of adding the struct definitions to the header, I would prefer adding (non-static/non-inline)  function ossl_ht_deref_value that would be implemented in hashtable.c and move this if inside it.